src/rules/no-subscribe-in-pipe.ts

```diff
@@ -0,0 +1,63 @@
+import { AST_NODE_TYPES, TSESTree as es } from '@typescript-eslint/utils';
+import { getTypeServices } from '../etc';
+import { ruleCreator } from '../utils';
+
+export const noSubscribeInPipeRule = ruleCreator({
+  defaultOptions: [],
+  meta: {
+    docs: {
+      description:
+        'Disallow calling of `subscribe` within any RxJS operator inside a `pipe`.',
+      recommended: 'recommended',
+      requiresTypeChecking: true,
+    },
+    fixable: undefined,
+    hasSuggestions: false,
+    messages: {
+      forbidden: 'Subscribe calls within pipe operators are forbidden.',
+    },
+    schema: [],
+    type: 'problem',
+  },
+  name: 'no-subscribe-in-pipe',
+  create: (context) => {
+    const { couldBeObservable, couldBeType } = getTypeServices(context);
+
+    function isWithinPipe(node: es.Node): boolean {
+      let parent = node.parent;
+
+      while (parent) {
+        if (
+          parent.type === AST_NODE_TYPES.CallExpression
+          && parent.callee.type === AST_NODE_TYPES.MemberExpression
+          && parent.callee.property.type === AST_NODE_TYPES.Identifier
+          && parent.callee.property.name === 'pipe'
+        ) {
+          return true;
+        }
+        parent = node.parent;
```

Comment:
@Danevandy99 the tests are stuck in an infinite loop.  This line should likely be `parent.parent`.
```suggestion
        parent = parent.parent;
```

Reply:
Ah, yep. Thanks for catching that.